service/history/task/task_test.go

```diff
@@ -280,6 +290,18 @@ func (s *taskSuite) TestHandleErr_ErrMaxAttempts() {
 	})
 }
 
+func (s *taskSuite) TestRetryErr() {
+	taskBase := s.newTestTask(func(task Info) (bool, error) {
+		return true, nil
+	}, nil)
+
+	s.Equal(false, taskBase.RetryErr(errWorkflowBusy))
+	s.Equal(false, taskBase.RetryErr(ErrTaskPendingActive))
+	s.Equal(false, taskBase.RetryErr(context.DeadlineExceeded))
+	s.Equal(false, taskBase.RetryErr(&redispatchError{Reason: "random-reason"}))
```

Comment:
this is a bit unexpected to see. What's the point of these NotEquals if there is the next line?

Reply:
added a comment